quilt/tools/store.py

```diff
@@ -30,43 +39,60 @@ class and its subclasses abstract file formats, file naming and
     BUILD_DIR = 'build'
     OBJ_DIR = 'objs'
     TMP_OBJ_DIR = os.path.join('objs', 'tmp')
+    PKG_DIR = 'pkgs'
+    VERSION = '1.0'
+    
+    def __init__(self, location=None):
+        if location is None:
+            location = default_store_location()
+            
+        assert os.path.basename(os.path.abspath(location)) == PACKAGE_DIR_NAME, \
+            "Unexpected package directory: %s" % location
+        self._path = location
+
+        objdir = os.path.join(self._path, self.OBJ_DIR)
+        tmpobjdir = os.path.join(self._path, self.TMP_OBJ_DIR)
+        pkgdir = os.path.join(self._path, self.PKG_DIR)
 
-    def __init__(self, store_dir=PACKAGE_DIR_NAME):
-        assert os.path.basename(os.path.abspath(store_dir)) == PACKAGE_DIR_NAME, \
-            "Unexpected package directory: %s" % store_dir
-        self._path = store_dir
+        if os.path.isdir(self._path):
+            # Verify existing package store is compatible
+            if self.VERSION != self._read_format_version():
```

Comment:
That’s right. This won’t help for users migrating from one version to the next, but it will ensure that users get notified if the client tries to access a quilt_packages directory that’s incompatible.